server/test/forms/FileUploadQuestionFormTest.java

```diff
@@ -52,4 +63,23 @@ public void getBuilder_withQdConstructor_returnsCompleteBuilder() throws Excepti
 
     assertThat(actual).isEqualTo(originalQd);
   }
+
+  @Test
+  public void setMaxFiles_emptyStringClearsMaxFiles() throws Exception {
+    FileUploadQuestionForm form = new FileUploadQuestionForm();
+    form.setMaxFiles("4");
+
+    assertThat(form.getMaxFiles().getAsInt()).isEqualTo(4);
+
+    form.setMaxFiles("");
+
+    assertThat(form.getMaxFiles()).isEmpty();
+  }
+
+  @Test
```

Comment:
suggestion (non-blocking):  Might be nice to write a test for a negative max files.

Reply:
Removed the negative approach altogether as per Jeff's comment.